site/src/pages/docs/client-factory-builder.mdx

```diff
@@ -0,0 +1,140 @@
+# Customizing a `ClientFactory` with `ClientFactoryBuilder`
+
+A <type://ClientFactoryBuilder> is a builder class that enables you to create a <type://ClientFactory>
+with custom options. For example, you might need to use the `connectTimeoutMillis` method to adjust the
+connection timeout of your clients.
+
+```java
+try (ClientFactory factory =
```

Comment:
How about removing try-with-resources because the factory shouldn't be closed until the client finishes sending/receiving requests/responses.

Reply:
This comment conflicts with [this comment](https://github.com/line/armeria/pull/3289#discussion_r559118603) from @ikhoon.

I think we need to decide which practice is more important than another comment. IMO, both opinions (try-with-resource to collect unnecessary resources and should not close until the client finishes its job) are crucial.

How about this? code example applies one opinion and left comment in above/below paragraph for another opinion ?

Reply:
Oops, I didn't see that comment. 😄 
I guess it's fine to use try-with-resources, then.
Let's just make sure to use `aggregate()` in the `try-with-resources` block.
Because if it's not aggregated, the request will be canceled right away. 😄 

Also, how about adding a section about the lifecycle of a factory? (e.g. A factory manages a connection pool and event loops so it should be closed after it's ...)?

Reply:
Recommend not using try/resources here since it's extremely rare to use it in real code, so it negatively effects the copy-pastability of this snippet.

Reply:
Yeah, a life cycle of a ClientFactory usually goes with an application's life cycle. 👍
It'd be better to remove try-with-resource in this example.

Reply:
Feel free to have a comment on https://github.com/line/armeria/pull/3289/commits/9f1a0f91e4508449ec655d6cd8cda188909eeadb :)

Reply:
Maybe you are more familiar than me on this context, or maybe I am poor at writing 😂